src/common.jl

```diff
@@ -87,15 +99,27 @@ const MultivariateDistribution{S<:ValueSupport} = Distribution{Multivariate,S}
 const MatrixDistribution{S<:ValueSupport}       = Distribution{Matrixvariate,S}
 const NonMatrixDistribution = Union{UnivariateDistribution, MultivariateDistribution}
```

Comment:
as mentioned on #941, this seems not required nor robust

Reply:
As I mentioned, this is not my code. There's lots of things I might change if I had freedom of action, but I didn't want to break anything as this is exported.

Reply:
this PR is going to be breaking anyway, I think we can remove this, it is defined and exported but not used anywhere else in the code base.

Reply:
I have gone to a lot of trouble to make it non-breaking - even with respect to `pdf()` still working with discrete distributions - but I'm willing to make it breaking if you think that's really okay?

Reply:
Breaking `NonMatrixDistribution` seems ok to me. That's why Distributions is still not 1.0, we are still allowing ourselves to fix the type system. If you want, you can add a deprecation for it

Reply:
No, it's fine. I'll delete it!